esphome/components/seeed_mr60fda2/seeed_mr60fda2.cpp

```diff
@@ -0,0 +1,437 @@
+#include "seeed_mr60fda2.h"
+#include "esphome/core/log.h"
+
+#include <utility>
+
+namespace esphome {
+namespace seeed_mr60fda2 {
+
+static const char *const TAG = "seeed_mr60fda2";
+
+// Prints the component's configuration data. dump_config() prints all of the component's configuration
+// items in an easy-to-read format, including the configuration key-value pairs.
+void MR60FDA2Component::dump_config() {
+  ESP_LOGCONFIG(TAG, "MR60FDA2:");
+#ifdef USE_BINARY_SENSOR
+  LOG_BINARY_SENSOR(" ", "People Exist Binary Sensor", this->people_exist_binary_sensor_);
+#endif
+#ifdef USE_BUTTON
+  LOG_BUTTON(" ", "Get Radar Parameters Button", this->get_radar_parameters_button_);
+  LOG_BUTTON(" ", "Reset Radar Button", this->reset_radar_button_);
+#endif
+#ifdef USE_SELECT
+  LOG_SELECT(" ", "Install Height Select", this->install_height_select_);
+  LOG_SELECT(" ", "Height Threshold Select", this->height_threshold_select_);
+  LOG_SELECT(" ", "Sensitivity Select", this->sensitivity_select_);
+#endif
+#ifdef USE_TEXT_SENSOR
+  LOG_TEXT_SENSOR(" ", "Is Fall Text Sensor", this->is_fall_text_sensor_);
+#endif
+}
+
+// Initialisation functions
+void MR60FDA2Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up MR60FDA2...");
+  this->check_uart_settings(115200);
+
+  this->current_frame_locate_ = LOCATE_FRAME_HEADER;
+  this->current_frame_id_ = 0;
+  this->current_frame_len_ = 0;
+  this->current_data_frame_len_ = 0;
+  this->current_frame_type_ = 0;
+  this->current_install_height_int_ = 0;
+  this->current_height_threshold_int_ = 0;
+  this->current_sensitivity_ = 0;
+  this->select_index_ = 0;
+
+  this->get_radar_parameters();
+
+  memset(this->current_frame_buf_, 0, FRAME_BUF_MAX_SIZE);
+  memset(this->current_data_buf_, 0, DATA_BUF_MAX_SIZE);
+
+  ESP_LOGCONFIG(TAG, "Set up MR60FDA2 complete");
+}
+
+// main loop
+void MR60FDA2Component::loop() {
+  uint8_t byte;
+
+  // Is there data on the serial port
+  while (this->available()) {
+    this->read_byte(&byte);
+    this->split_frame_(byte);  // split data frame
+  }
+}
+
+/**
+ * @brief Calculate the checksum for a byte array.
+ *
+ * This function calculates the checksum for the provided byte array using an
+ * XOR-based checksum algorithm.
+ *
+ * @param data The byte array to calculate the checksum for.
+ * @param len The length of the byte array.
+ * @return The calculated checksum.
+ */
+uint8_t MR60FDA2Component::calculate_checksum_(const uint8_t *data, size_t len) {
```

Comment:
This doesn't need to be a member of the class and can be static outside it.
```suggestion
static uint8_t calculate_checksum(const uint8_t *data, size_t len) {
```